src/linker/Linker.Steps/OutputStep.cs

```diff
@@ -103,7 +103,34 @@ protected virtual void WriteAssembly (AssemblyDefinition assembly, string direct
 				}
 			}
 
-			assembly.Write (GetAssemblyFileName (assembly, directory), writerParameters);
+			writerParameters.DeterministicMvid = true;
+
+			var stream = new MemoryStream ();
+
+			var fileName = GetAssemblyFileName (assembly, directory);
+			assembly.Write (stream, writerParameters);
+
+			//
+			// Check whenever the output exits and has the same guid
+			//
+			var prevGuid = Guid.Empty;
+			if (File.Exists (fileName)) {
+				try {
+					var prevModule = ModuleDefinition.ReadModule (fileName);
+					prevGuid = prevModule.Mvid;
+				} catch {
+				}
+			}
+			if (assembly.MainModule.Mvid != prevGuid) {
+				if (Context.LogMessages)
```

Comment:
I believe there is something like  Context.Logger.LogMessage that should be used instead of console write line.